crates/router/src/connector/payu/transformers.rs

```diff
@@ -0,0 +1,550 @@
+        let auth_type = PayuAuthType::try_from(&item.connector_auth_type)?;
+        let payment_method = match item.request.payment_method_data.clone() {
+            api::PaymentMethod::Card(ccard) => Ok(PayuPaymentMethod {
+                pay_method: PayuPaymentMethodData::Card(PayuCard::Card {
+                    number: ccard.card_number.peek().clone(),
+                    expiration_month: ccard.card_exp_month.peek().clone(),
+                    expiration_year: ccard.card_exp_year.peek().clone(),
+                    cvv: ccard.card_cvc.peek().clone(),
+                }),
+            }),
+            api::PaymentMethod::Wallet(wallet_data) => match wallet_data.issuer_name {
+                api_models::enums::WalletIssuer::GooglePay => Ok(PayuPaymentMethod {
+                    pay_method: PayuPaymentMethodData::Wallet({
+                        PayuWallet {
+                            value: "ap".to_string(),
+                            typo: "PBL".to_string(),
+                            authorization_code: wallet_data
+                                .token
+                                .get_required_value("token")
+                                .change_context(errors::ConnectorError::RequestEncodingFailed)
+                                .attach_printable("No token passed")?,
+                        }
+                    }),
+                }),
+                api_models::enums::WalletIssuer::ApplePay => Ok(PayuPaymentMethod {
+                    pay_method: PayuPaymentMethodData::Wallet({
+                        PayuWallet {
+                            value: "jp".to_string(),
+                            typo: "PBL".to_string(),
+                            authorization_code: wallet_data
+                                .token
+                                .get_required_value("token")
+                                .change_context(errors::ConnectorError::RequestEncodingFailed)
+                                .attach_printable("No token passed")?,
+                        }
+                    }),
+                }),
+                _ => Err(errors::ConnectorError::NotImplemented(
+                    "Unknown Wallet in Payment Method".to_string(),
+                )),
+            },
+            _ => Err(errors::ConnectorError::NotImplemented(
+                "Unknown payment method".to_string(),
+            )),
+        }?;
+        Ok(Self {
+            customer_ip: "127.0.0.1".to_string(), //todo take input from core
+            merchant_pos_id: auth_type.merchant_pos_id,
+            total_amount: item.request.amount.to_string(),
+            currency_code: item.request.currency.to_string(),
+            description: item.description.clone().ok_or(
+                errors::ConnectorError::MissingRequiredField {
+                    field_name: "item.description".to_string(),
+                },
+            )?,
```

Comment:
Does PayU require a description to be always present? If not, wrap the field in an `Option`.

Reply:
Yes, Description is always required by Payu